### PR TITLE
New mecha module - Engineering toolbox

### PR DESCRIPTION
**Pull request by ErdGinalD**
## What Does This PR Do
Add a new module to engineering mechas - toolbox. He contains all regular tools for primary engineer with x2 speed modifyer. Steal your NanoMed now!
With emag, revealing 25 damage knife, which is copypasted behavior of toolbelt implant, but due to "this is mecha", it slightly better than combat knife.

According to https://github.com/ParadiseSS13/Paradise/pull/23802 previous attempt, welding tool now doesnt self-refill, but have a faster interaction + lower energy consumption

Also rename all proc check_menu into open_radial_menu for better understanding.

## Why It's Good For The Game
More useful modules for engineering mechas, more funny things happened because of that. 

## Images of changes
image
image
after emag, we have machete
image

## Testing
Copy code from (https://github.com/ParadiseSS13/Paradise/pull/23802), cut the not needed code, all work perfectly in local server.

<hr>

### Declaration
- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/feature_freeze/CODE_OF_CONDUCT.md#currently-changes-to-the-following-types-of-content-reuires-pre-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.

<hr>

## Changelog
:cl:
add: Added toolbox module for mecha, who contain x2 faster toolbelt. With emag, reveal 25 damage machete for tator shenanigans.
/:cl:

### CODE REVIEW

**Comment by SynthTwo** [comment]
!wiki_label

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 1]
```diff
@@ -455,6 +455,99 @@
...
+	drop_sound = 'sound/items/handling/toolbox_drop.ogg'
+	pickup_sound =  'sound/items/handling/toolbox_pickup.ogg'
+	force = 10
+	equip_cooldown = 0.8 SECONDS
+	energy_drain = 50
+	harmful = TRUE
+	var/list/items_list = newlist(/obj/item/screwdriver/cyborg, /obj/item/wrench/cyborg, /obj/item/weldingtool/mecha,
+		/obj/item/crowbar/cyborg, /obj/item/wirecutters/cyborg, /obj/item/multitool/cyborg) //0.5 toolspeed all
+	var/obj/item/selected_item
+	var/emag_item = /obj/item/kitchen/knife/combat/cyborg/mecha
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/tools/work_tools.dm:475]: document these vars please

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 2]
```diff
@@ -455,6 +455,99 @@
...
+/obj/item/mecha_parts/mecha_equipment/eng_toolset/New()
+	..()
+	for(var/obj/item/item as anything in items_list)
+		item.flags |= NODROP
+		item.resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+		item.slot_flags = null
+		item.w_class = WEIGHT_CLASS_HUGE
+		item.materials = null
+		item.tool_enabled = TRUE
+	selected_item = pick(items_list)
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/tools/work_tools.dm:486]: put this in `Initialize()` please

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 3]
```diff
@@ -455,6 +455,99 @@
...
+		else
+			. += "|<a href='byond://?src=[UID()];select=[item.UID()]'>[short_name]</a>"
+	. += "|"
+
+/obj/item/mecha_parts/mecha_equipment/eng_toolset/Topic(href,href_list)
+	..()
+	if(href_list["select"])
+		selected_item = locateUID(href_list["select"])
+		if(!istype(selected_item))
+			return
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/tools/work_tools.dm:503]: you need to sanity check this selected_item. If i inject an href here, I could put in a UID for something that is not in our mecha_equipment. Check to make sure this is an item you are supposed to be able to select.

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 4]
```diff
@@ -455,6 +455,99 @@
...
+	if(isliving(target))
+		chassis.do_attack_animation(target)
+	chassis.use_power(energy_drain)
+	return TRUE
+
+/obj/item/mecha_parts/mecha_equipment/eng_toolset/self_occupant_attack()
+	radial_menu(chassis.occupant)
+
+/obj/item/mecha_parts/mecha_equipment/eng_toolset/proc/check_menu(mob/living/carbon/user)
+	return (user && chassis.occupant == user && user.stat != DEAD)
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/tools/work_tools.dm:520]: document this proc please

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 5]
```diff
@@ -455,6 +455,99 @@
...
+
+/obj/item/mecha_parts/mecha_equipment/eng_toolset/proc/radial_menu(mob/living/carbon/user)
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/tools/work_tools.dm:522]: rename this to something like `open_radial_menu` which indicates the action it's trying to perform

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 6]
```diff
@@ -455,6 +455,99 @@
...
+		return
+	var/obj/item/selected
+	for(var/obj/item/item as anything in items_list)
+		if(item.name == choice)
+			selected = item
+			break
+	if(selected)
+		extend(selected)
+
+/obj/item/mecha_parts/mecha_equipment/eng_toolset/proc/extend(obj/item/selected)
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/tools/work_tools.dm:537]: document proc

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

**Comment by lewcc** [comment]
This better have a plain "toolbox" module just for bashing people with

**Comment by ErdGinalD** [comment]
> This better have a plain "toolbox" module just for bashing people with

You can bash it with this module in hand, who act exactly like green toolbox!

File: code/game/mecha/equipment/tools/work_tools.dm (hunk 5 above)
Comment (ErdGinalD) [reply code/game/mecha/equipment/tools/work_tools.dm:522]: Renamed all procs check_menu into open_radial_menu

**Review by Sirryan2002 — APPROVED** [review]

**Comment by dudley500** [comment]
Hello @ErdGinalD could you please provide any info you have relating to the Pre-approval that's required for any antag/security content mentioned [here](https://discord.com/channels/145533722026967040/499000353312735233/1271125308975878175)

**Comment by S34NW** [comment]
(Specifically regarding the emag effect, the rest of the PR is fine)

**Comment by Denthamos** [comment]
Not sure its a good idea to have tool module for mechs that anyone can use.  Maybe if there was department centric mechs that would have default access tied to their department and robotics if when created it was not set to be open using an access circuit that would be used to make mechs (Something I was talking about for a possible future thing with another design member).  Then such modules could only be used on that mech and not more common use ones like ripley's.

**Comment by FunnyMan3595** [comment]
Doesn't make sense to me. A mecha is a big clunky machine, it doesn't have the dexterity to use precision tools. The RCD is reasonable because it's a big thing itself, and you only really need to point it at the right tile. But using a screwdriver? I'm not buying it.

**Comment by ErdGinalD** [comment]
> Doesn't make sense to me. A mecha is a big clunky machine, it doesn't have the dexterity to use precision tools. The RCD is reasonable because it's a big thing itself, and you only really need to point it at the right tile. But using a screwdriver? I'm not buying it.

Most reason is to do more actions not getting out of mecha. Because ripley is engineering mecha too, he freely can wear tools. This is useful, but overall not criticical module. I just tired from "get in mecha, goes to somewhere, get out, click 1 time with tool, get in". This is lame. Toolbox avaliable on other servers and dont do some revolution, just handy tool for average usage.

**Comment by ErdGinalD** [comment]
> Not sure its a good idea to have tool module for mechs that anyone can use. Maybe if there was department centric mechs that would have default access tied to their department and robotics if when created it was not set to be open using an access circuit that would be used to make mechs (Something I was talking about for a possible future thing with another design member). Then such modules could only be used on that mech and not more common use ones like ripley's.

Ripley is specifically engineering/mining mecha, he meant to be in usage by engineers or miners. If he get in other hands - this is not module problem, like chief stoling durand for example.

**Comment by githubuser4141** [comment]
> Doesn't make sense to me. A mecha is a big clunky machine, it doesn't have the dexterity to use precision tools. The RCD is reasonable because it's a big thing itself, and you only really need to point it at the right tile. But using a screwdriver? I'm not buying it.

Why wouldn't it? It's an attachment, a pod full of tools that the mech's arm holds up to the general area, then the smaller, more precise tools in the POD extend and do their work. The mech's arm doesn't hold anything, it's just a pod where each tube contains a single tool, so 6 or 7 tubes, each tube has a little sensor to tell the mech's arm to move where it needs to be positioned, then the driver uses a controller to guide the tool to the work site.

**Comment by githubuser4141** [comment]
> Not sure its a good idea to have tool module for mechs that anyone can use. Maybe if there was department centric mechs that would have default access tied to their department and robotics if when created it was not set to be open using an access circuit that would be used to make mechs (Something I was talking about for a possible future thing with another design member). Then such modules could only be used on that mech and not more common use ones like ripley's.

Anyone can already use a autolathe to make tools, or get tools from a vendor or a toolbox, or get someone to print advanced tools from RND. 

If you have access to building mechs you can do more dangerous things with them anyone, a mecha RCD is more dangerous then a mecha toolkit in 90% of situations, you can cause breaches, spam walls in security, break into most places, use it for combat in a 1x1 hallway by putting a wall behind them and drilling/clamping them as they can no longer run away. This is a mid-round tool that still requires materials and research and is locked to being on a slow, clunky ass mech.

**Comment by ErdGinalD** [comment]
> The only thing that's a bit weird is the machete

This is obvious reference to toolset implant, who contain combat knife after emagging. Tookit module is in fact a toolset implant for peoples, but slightly bigger. Combat knife is also slightly bigger.

**Comment by ErdGinalD** [comment]
Closed due to https://github.com/ParadiseSS13/Paradise/pull/26550 exist.

**Pull request closed without merging**